tasks/utils.py

```diff
@@ -376,19 +377,55 @@ def query_version(ctx, git_sha_length=7, prefix=None, major_version_hint=None):
     return version, pre, commit_number, git_sha, pipeline_id
 
 
+def cache_version(ctx, git_sha_length=7, prefix=None):
+    """
+    Generate a pickle cache file containing all needed variables used by get_version.
+    """
+    packed_data = []
+    for maj_version in ['6', '7']:
+        version, pre, commits_since_version, git_sha, pipeline_id = query_version(
+            ctx, git_sha_length, prefix, major_version_hint=maj_version
+        )
+        packed_data.append((version, pre, commits_since_version, git_sha, pipeline_id))
+    packed_data.append(is_allowed_repo_nightly_branch(os.getenv("BUCKET_BRANCH")))
+    with open("_version.cache", "wb") as file:
+        pickle.dump(packed_data, file)
+
+
 def get_version(
     ctx, include_git=False, url_safe=False, git_sha_length=7, prefix=None, major_version='7', include_pipeline_id=False
 ):
-    # we only need the git info for the non omnibus builds, omnibus includes all this information by default
-
     version = ""
-    version, pre, commits_since_version, git_sha, pipeline_id = query_version(
-        ctx, git_sha_length, prefix, major_version_hint=major_version
-    )
+    pipeline_id = os.getenv("CI_PIPELINE_ID")
+    if pipeline_id is not None:
```

Comment:
This would also filter empty strings:
```suggestion
    if pipeline_id:
```